### DISCUSSION TITLE
Multiple objectives using blocking constraints

### DISCUSSION THREAD

**Discussion by speedfreeman** [google/or-tools#4183] [CP-SAT questions] [answered]
The guidance for solving multiple lexicographically ordered objectives seems to be:

Call the solver to obtain solution _s_ for the optimal value _v_ of objective _i_
Add a constraint that objective _i_ == _v_
Add _s_ as a solution hint
Repeat for objective _i+1_

But what about not adding a solution hint for _s_ but instead adding a constraint to block _s_? If the solver returns infeasible for anything other than the first objective, then, of the solutions obtained so far, those that are best for all objectives must also be optimal for objectives _i+1_ to _n_?

**Comment by lperron** [comment, marked as answer]
you can. It is just harder for the solver at each iterations as the model is bigger (less fixed variables).
